functions/index.js

```diff
@@ -89,6 +103,9 @@ exports.dialogflowFirebaseFulfillment = functions.https.onRequest((request, resp
 
   // Run the proper function handler based on the matched Dialogflow intent name
   let intentMap = new Map();
-  intentMap.set('QueryHandler', WebCallIntent);
+  //using exercise entity to decise which query handler to call
+  var exercise = agent.parameters["exercise"];
+  if(exercise == "exercise") intentMap.set('ExerciseQueryHandler',WebCallExercise);
```

Comment:
Rather than do an if-else here, you can just add `intentMap.set('ExerciseQueryHandler',WebCallExercise)` to the intentMap.  The `agent.handleRequest(intentMap)` will select the correct one based on the input. 👍

Reply:
Thanks! Wasn't sure on that